### PR TITLE
WT-10104 Find the right executable in print_stack_trace.py

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: test/evergreen/print_stack_trace.py [hunk 1]
```diff
@@ -143,6 +143,13 @@ def main():
         start = output.find('execfn: ')
         # Fetch the value of execfn. This is the executable path!
         executable_path = re.search(r'\'.*?\'', output[start:]).group(0)
+        executable_path = executable_path.replace("'", "" )
+
+        # If the core dump comes from a Python test, we don't need to construct the executable path.
+        if "python" not in executable_path.lower():
+            # The executable is where the core dump is.
+            dirname = core_file_path.rsplit('/', 1)[0]
```
Comment (Siddhartha8899) [inline test/evergreen/print_stack_trace.py:151]: what if the core is in the current dir?

Comment (Siddhartha8899) [reply test/evergreen/print_stack_trace.py:151]: In other words, if the `core_file_path` is `dump_blah`, then the dirname would be `dump_blah` as well, no?

Comment (Siddhartha8899) [reply test/evergreen/print_stack_trace.py:151]: My bad the `core_file_path` will be `./dump_blah`, so it shouldn't be an issue.

**Review by Siddhartha8899 — APPROVED** [review]
LGTM! Thanks for fixing this!

File: test/evergreen/print_stack_trace.py (hunk 1 above)
Comment (etienneptl) [reply test/evergreen/print_stack_trace.py:151]: Yes, tested again and it works:

From the `build` dir:
```
> ./test/format/t
program name not set: FAILED: main/197: false
Segmentation fault (core dumped)
> python3 ../test/evergreen/print_stack_trace.py
+---------------------+
| ./dump_t.22086.core |
+---------------------+
executable_path ./test/format/t
```

From the `build/test/format` dir:
```
./t
program name not set: FAILED: main/197: false
Segmentation fault (core dumped)
python3 ../../../test/evergreen/print_stack_trace.py
+---------------------+
| ./dump_t.22362.core |
+---------------------+
executable_path ./t
```

In both cases, the executable path is constructed corrected.

**Pull request merged**